Implement data generator for odin

tbukovics opened the pull request:
### Issue reference / description

The branch you merge from should already reference an event-formation-unit github ticket number. You can add a descriptive title, but if an issue is referenced, you don't have to.

## Checklist for submitter

- [x] Check for conflict with integration test
- [x] Unit tests pass

---

### Nominate for Group Code Review

- [x] Nominate for code review

mortenjc commented on src/modules/timepix3/generators/odin_datagen.bash:
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Parse command line arguments
+while getopts "f:d" opt; do
```
This is pretty old school :-) Are you familiar with argparse?

import argparse

https://docs.python.org/3/library/argparse.html

It is a bit more flexible IMO

mortenjc commented on src/modules/timepix3/generators/odin_datagen.bash:
```diff
@@ -0,0 +1,57 @@
+fi
+
+# Define a function to be executed on exit
+cleanup() {
```
I would add

function cleanup() {
}

mortenjc commented on src/modules/timepix3/generators/tpx_generator.py:
```diff
@@ -0,0 +1,240 @@
+TDC_TRIGGERCOUNTER_MASK = 0x00FFF00000000000
+TDC_TIMESTAMP_MASK = 0x00000FFFFFFFFE00
+TDC_STAMP_MASK = 0x00000000000001E0
+TDC_TYPE_OFFSET = 56
```
```
When dealing with masks I definitely prefer to align the fields 
TDC_TRIGGERCOUNTER_MASK =  0x00FFF00000000000
TDC_TIMESTAMP_MASK =       0x00000FFFFFFFFE00
TDC_STAMP_MASK =           0x00000000000001E0
```

tbukovics commented on src/modules/timepix3/generators/tpx_generator.py:
```diff
@@ -0,0 +1,240 @@
+
+                    data = chunk[tpx_mask]
+
+                    packet = Ether() / IP(dst="10.103.2.33") / UDP(sport=4096, dport=9888) / data.tobytes()
```
agree

tbukovics commented on src/modules/timepix3/generators/tpx_generator.py:
```diff
@@ -0,0 +1,240 @@
+        self.period_ns = int(1 / self.frequency_hz * 1e9)
+        self.send_process = None
+        self.packet_time = time.time()
+        self.udp_dport = 9888
+        self.udp_sport = 4096
```
aggree

tbukovics commented:
@morten and @amues 

I requested a new review, since I reworked a bit the python file. I also removed the bash, as we discussed with @amues this python will be delivered with EFU generators and the python env will be configured from ansible.

I modified the CMAKE to have a new script which can copy over a python file structure into the defined runtime folder, and set executeable permissions. So the python file should be moved to the build/generators by CMAKE
